asistente_ladm_col/logic/quality/line_quality_rules.py

```diff
@@ -0,0 +1,731 @@
+            pr.addAttributes(rule.error_table_fields)
+            error_layer.updateFields()
+
+            end_points, dangle_ids = self.get_dangle_ids(boundary_layer)
+
+            new_features = []
+            for dangle in end_points.getFeatures(dangle_ids):
+                new_feature = QgsVectorLayerUtils().createFeature(end_points,
+                                                                  dangle.geometry(),
+                                                                  {0: dangle[db.names.T_ILI_TID_F],
+                                                                   1: self.quality_rules_manager.get_error_message(QUALITY_RULE_ERROR_CODE_E200501),
+                                                                   2: QUALITY_RULE_ERROR_CODE_E200501})
+                new_features.append(new_feature)
+
+            error_layer.dataProvider().addFeatures(new_features)
+
+            if error_layer.featureCount() > 0:
+                added_layer = self.qgis_utils.add_error_layer(db, error_layer)
+
+                return (QCoreApplication.translate("LineQualityRules",
+                                 "A memory layer with {} boundary dangles has been added to the map!").format(added_layer.featureCount()), Qgis.Critical)
+
+            else:
+                return (QCoreApplication.translate("LineQualityRules",
+                                 "Boundaries have no dangles!"), Qgis.Success)
+
+
+    # UTILITY METHODS
+    def get_boundary_nodes_features_not_covered_by_boundary_points(self, db, boundary_point_layer, boundary_layer, point_bfs_layer, error_layer, id_field):
+        dict_uuid_boundary_point = get_uuid_dict(boundary_point_layer, db.names, db.names.T_ID_F)
+        dict_uuid_boundary = get_uuid_dict(boundary_layer, db.names, db.names.T_ID_F)
+        tmp_boundary_nodes_layer = processing.run("native:extractvertices", {'INPUT': boundary_layer, 'OUTPUT': 'memory:'})['OUTPUT']
+
+        # layer is created with unique vertices, it is necessary because 'remove duplicate vertices' processing algorithm does not filter the data as we need them
+        boundary_nodes_unique_layer = QgsVectorLayer("Point?crs={}".format(boundary_layer.sourceCrs().authid()), 'unique boundary nodes', "memory")
+        data_provider = boundary_nodes_unique_layer.dataProvider()
+        data_provider.addAttributes([QgsField(id_field, QVariant.Int)])
+        boundary_nodes_unique_layer.updateFields()
+
+        id_field_idx = tmp_boundary_nodes_layer.fields().indexFromName(id_field)
+        request = QgsFeatureRequest().setSubsetOfAttributes([id_field_idx])
+
+        filter_fs = list()
+        fs = list()
+        for f in tmp_boundary_nodes_layer.getFeatures(request):
+            item = [f[id_field], f.geometry().asWkt()]
+            if item not in filter_fs:
+                filter_fs.append(item)
+                fs.append(f)
+        boundary_nodes_unique_layer.dataProvider().addFeatures(fs)
```

Comment:
Isn't this block what `geometry.get_polygon_nodes_layer() does? We should unify them into a single function. (TODO: for a follow-up PR).